src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/IncludeExpression.cs

```diff
@@ -0,0 +1,72 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Models;
+
+namespace Microsoft.Health.Fhir.Core.Features.Search.Expressions
+{
+    /// <summary>
+    /// Represents an include expression (where an additional resource is included based on a reference)
+    /// </summary>
+    public class IncludeExpression : Expression
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="IncludeExpression"/> class.
+        /// </summary>
+        /// <param name="resourceType">The resource that supports the reference.</param>
+        /// <param name="referenceSearchParameter">THe search parameter that establishes the reference relationship.</param>
+        /// <param name="targetResourceType">The target type of the reference.</param>
+        /// <param name="wildCard">If this is a wildcard reference include (include all referenced resources).</param>
+        public IncludeExpression(string resourceType, SearchParameterInfo referenceSearchParameter, string targetResourceType, bool wildCard)
+        {
+            EnsureArg.IsNotNullOrWhiteSpace(resourceType, nameof(resourceType));
+
+            if (!wildCard)
```

Comment:
Could consider the include expression to be wildcard if `targetResourceType` is null?

Reply:
Nevermind, I guess that's used to specifically scoping down to a particular type when the reference can point to various resource types.